packages/react-inbox/src/reducer.ts

```diff
@@ -0,0 +1,49 @@
+
+export default (state, action) => {
+  switch (action.type) {
+    case "inbox/INIT": {
+      return {
+        ...state,
+        config: action.payload.config,
+      };
+    }
+
+    case "inbox/SET_LOADING": {
+      return {
+        ...state,
+        isLoading: action.payload.isLoading,
+      };
+    }
+
+    case "inbox/SET_MESSAGES": {
+      return {
+        ...state,
+        messages: action?.payload?.messages?.nodes?.map(message => ({
+          messageId: message.messageId,
+          created: message.created,
+          title: message?.content?.title,
+          body: message?.content?.body,
+          data: message?.content?.data,
+        })),
+      };
+    }
+
+    case "inbox/NEW_MESSAGE": {
+      return {
+        ...state,
+        messages: [
+          {
+            created: new Date().getTime(),
+            messageId: action.payload.messageId ?? new Date().getTime(),
```

Comment:
we don't pass messageId through the websocket but we need some kind of id for react key so using datetime for now

Reply:
what happens if there is any action/side-effect associated with this? Say mark read or delete? We would need an id to do anything with this right?

Reply:
yeah, the way it works right now is that we have tracking links in action.payload.data that will fire off... but that isn't all hooked up yet